kubernetes/controller/internal/resolution/workerpool/workerpool.go

```diff
@@ -72,13 +85,22 @@ func NewWorkerPool(opts PoolOptions) *WorkerPool {
 		opts.QueueSize = 100
 	}
 
+	const eventChannelSize = 1000
 	return &WorkerPool{
 		PoolOptions: opts,
 		workQueue:   make(chan *WorkItem, opts.QueueSize),
-		inProgress:  make(map[string]struct{}),
+		// TODO: I bet Jakob will tell me to use an Informer.
+		inProgress: make(map[string][]RequesterInfo),
```

Comment:
why do you need a map to requester info? do you use this to fan out events again?